pkg/gatewayserver/upstream/ns/ns.go

```diff
@@ -65,8 +66,19 @@ func (h *Handler) ConnectGateway(ctx context.Context, ids ttnpb.GatewayIdentifie
 	if conn.Frontend().SupportsDownlinkClaim() {
 		return nil
 	}
-	h.cluster.ClaimIDs(ctx, ids)
-	defer h.cluster.UnclaimIDs(ctx, ids)
+	logger := log.FromContext(ctx)
```

Comment:
This claims downlinks and not gateways. Logic looks good to me. Please fix the log messages.